bin/DataProcess.pm

```diff
@@ -335,110 +335,119 @@ sub fastq2tab_pe_java {
 	use File::Basename;
 	my $input_dir =  $args_dict->{'random_id'};
 	my $input_dir1 =  "/$args_dict->{'random_id'}_input/fastq_paired_end";
-	
+
+	my $readtools = $args_dict{'readtools'};
+
 	my $hdfs_exe = $args_dict->{"hdfs_exe"};
-	
+
 	if (system ("$hdfs_exe dfs -test -d $input_dir1")==0) {
 		system("$hdfs_exe dfs -rm -r $input_dir1") == 0 || warn "Error in deleting the $input_dir1 folder on hdfs file system. $input_dir1 does not exists\n";
 	}
-	
+
 	my $script_current_directory = abs_path($0);
 	my $current_directory = cwd();
-	
+
 	my $script_dirname  = dirname($0);
 	#my $jar_file_path = "$script_dirname/bin/JDistmap.jar";
-	my $jar_file_path = "$script_dirname/bin/ReadTools-1.0.0.jar";
-	
+	# ReadTools v1.0+ is too large to be distributed with distmap
+	#my $jar_file_path = "$script_dirname/bin/ReadTools-1.0.0.jar";
+
 	my $block_size = $args_dict->{"block_size"};
-	
+
 	$block_size =~ s/\D//g;
 	my $block_size_byte = $block_size*1024*1024;
-	
+
 
 	my $file_count = scalar(@$file_list);
 	$file_count = $file_count+1;
 	my $output_dir = "$args_dict->{'output_directory'}/$args_dict->{'random_id'}/$args_dict->{'fastq_dir_pe'}";
 	my $index=1;
-        my $xp = qw(/configuration/property[string(name)="fs.defaultFS"]/value);
-        my $hdfs = XML::XPath->new(filename => "$ENV{HADOOP_CONF_DIR}/core-site.xml")->findvalue($xp);
+    # my $xp = qw(/configuration/property[string(name)="fs.defaultFS"]/value);
+    # my $hdfs = XML::XPath->new(filename => "$ENV{HADOOP_CONF_DIR}/core-site.xml")->findvalue($xp);
+	my $hdfs = `$hdfs_exe getconf -confKey "fs.defaultFS"`;
 	foreach my $f (@$file_list) {
 		my $output_file = "$output_dir/fastq_file".$index;
 		my $read1_fastq=$f->[0];
 		my $read2_fastq=$f->[1];
-		
+
 		#my $cmd = "java -Xmx4g -jar $jar_file_path UploadFastq HADOOP_HOME=$args_dict->{'hadoop_home'} I=$read1_fastq I=$read2_fastq O=file:///$input_dir BZ2=true BLOCK_SIZE=$block_size_byte MAPPER=BWA";
-		my $cmd = "java -Xmx4g -jar $jar_file_path ReadsToDistmap --input $read1_fastq --input2 $read2_fastq --output $hdfs/$input_dir1/fastq_file1.bz2 --hdfsBlockSize $block_size_byte";
-		
+		#my $cmd = "java -Xmx4g -jar $jar_file_path ReadsToDistmap --input $read1_fastq --input2 $read2_fastq --output $hdfs/$input_dir1/fastq_file1.bz2 --hdfsBlockSize $block_size_byte";
+		my $cmd = "JAVA_OPTS=-Xmx4g $readtools ReadsToDistmap --input $read1_fastq --input2 $read2_fastq --output $hdfs/$input_dir1/fastq_file1.bz2 --hdfsBlockSize $block_size_byte";
```

Comment:
Several things here:
* If the binary is the jar file, will it work properly with the `JAVA_OPTS`? I think that it will use the ones provided in the `$readtools` options...
* For the future, this can support FASTQ and also SAM/BAM/CRAM inputs. Maybe an issue might be open...

Reply:
Open #15 for the format stuff